samples/authUserFlow.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const {OAuth2Client} = require('google-auth-library');
+const puppeteer = require('puppeteer');
+const url = require('url');
+const http = require('http');
+
+// Download your OAuth2 configuration from the Google
+const keys = require('./oauth2.keys.json');
+
+/**
+ * Keep a config.keys.json with a username and password
+ * that correspond to your test google account:
+ * {
+ *   "username": "your-user-name@gmail.com",
+ *   "password": "your-password"
```

Comment:
Is this necessary? I feel uncomfortable putting my password in such a file. Isn't it possible to post a link to `https://accounts.google.com/o/oauth2/auth?client_id=...` and then wait for user input on the authentication code?